### ISSUE TITLE
Non-Interactive Bloom

### ISSUE THREAD

**Issue by danthony06** [ros-infrastructure/bloom#667]
Is it possible to run ```bloom-release``` non-interactively, i.e. completely specify all parameters through the command line so I don't have manually step through things during the process? I'm looking at how to integrate ```bloom``` into our CI/CD pipelines, and so far it's been tough since it seems to require manually interacting with the process.

**Comment by cottsay** [comment]
It's possible - bloom's behavior is inconsistent right now, but we're working on making that story better.

#668 is part of that effort.

**Comment by nuclearsandwich** [comment]
Running `bloom-release` with its `--non-interactive` argument will work for many situations but there are some know restrictions which I'll document here to start and then as part of work on this issue we'll improve the real documentation. I don't wonder if Scott can think of a few more cases.

* If the track is new, interactivity is required to fill out the new track
* If creating the first release for a repository in a given rosdistro, interactivity will be required to create the PR.
* If the release repository has custom patches, and those fail to apply, interactivity _was_ required and use in non-interactive contexts would have resulted in misapplied patches being mistakenly marked as resolved. (Fixed by #668)
* If rosdep keys are missing for a package, rodsep will prompt for a refresh ignoring non-interactivity.

> completely specify all parameters through the command line

I'm not sure if a command-line is the best place to prepopulate answers to bloom's prompts or if it's better to encourage updating the configuration (both in the rosdistro and release repository's tracks.yaml) ahead of running in a non-interactive context. But that still leaves the "bootstrapping" / first-run problem in place.

**Comment by danthony06** [comment]
Yeah, I poked around a little more and found the limitations. I'm mostly looking at tying bloom to a GitHub action, so that when we do a release, I can get the action to release the update to the buildfarm.

**Comment by nuclearsandwich** [comment]
@danthony06 please feel free to note any additional stumbling blocks here so that we can keep track of them as we improve similar workflows.
